test/powershell/Modules/Microsoft.PowerShell.Diagnostics/Import-Counter.Tests.ps1

```diff
@@ -0,0 +1,433 @@
+<############################################################################################
+ # File: Import-Counter.Tests.ps1
+ # Provides Pester tests for the Import-Counter cmdlet.
+ ############################################################################################>
+
+# Translate a counter set name to a localized name
+function TranslateCounterSetName($counterName)
+{
+    $id = Get-PerformanceCounterID $counterName
+    if ($id -ne $null)
+    {
+        $translatedName = Get-PerformanceCounterLocalName $id
+        if ($translatedName -ne $null)
+        {
+            return $translatedName
+        }
+    }
+
+    return $counterName
+}
+
+Describe "Tests for Import-Counter cmdlet" -Tags "CI" {
+
+    BeforeAll {
+        $cmdletName = "Import-Counter"
+
+        . "$PSScriptRoot/CounterTestHelperFunctions.ps1"
+
+        $counterPaths = @(
+            (TranslateCounterPath("\Memory\Available Bytes"))
+            (TranslateCounterPath("\processor(*)\% Processor time"))
+            (TranslateCounterPath("\Processor(_Total)\% Processor Time"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Current Disk Queue Length"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Disk Bytes/sec"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Disk Read Bytes/sec"))
+        )
+        $setNames = @{
+            Memory = (TranslateCounterSetName("memory"))
+            PhysicalDisk = (TranslateCounterSetName("physicaldisk"))
+            Processor = (TranslateCounterSetName("processor"))
+        }
+
+        $outputDirectory = $PSScriptRoot
+        $rootFilename = "exportedCounters"
+        $blgPath = "$outputDirectory/$rootFilename.blg"
+        $csvPath = "$outputDirectory/$rootFilename.csv"
+        $tsvPath = "$outputDirectory/$rootFilename.tsv"
+        $badSamplesBlgPath = "$outputDirectory/assets/BadCounterSamples.blg"
+        $corruptBlgPath = "$outputDirectory/assets/CorruptBlg.blg"
+        $notFoundPath = "$outputDirectory/DAD288C0-72F8-47D3-8C54-C69481B528DF.blg"
+
+        Write-Host "Gathering counter values for export..."
```

Comment:
I put that in only because the gathering of counter data takes 25 seconds to run. I'll go ahead and remove it.